### DISCUSSION TITLE
CVE-2024-3924 False Positive (text-generation)

### DISCUSSION THREAD

**Discussion by muhammad-asn** [aquasecurity/trivy#6873] [False Detection]
### IDs

CVE-2024-3924

### Description

We got text-generation CVE-2024-3924 although the library not installed.
Also from the pypi the latest package in 0.7 
https://pypi.org/project/text-generation/#history

### Reproduction Steps

```bash
1. Install package
2. Copy in the docker image
3. Run trivy scanning
```

### Target

Container Image

### Scanner

Vulnerability

### Target OS

-

### Debug Output

```bash
$ trivy image \
      --cache-dir /root/.cache/trivy \
      --severity MEDIUM,HIGH,CRITICAL \
      --exit-code 1 \
      --no-progress \
      --ignore-unfixed \
      --scanners vuln \
      backend-image:latest
```

### Version

```bash
$ trivy --version
Version: 0.51.4
```

### Checklist

- [X] Read [the documentation regarding wrong detection](https://aquasecurity.github.io/trivy/dev/community/contribute/discussion/#false-detection)
- [X] Ran Trivy with `-f json` that shows data sources and confirmed that the security advisory in data sources was correct

**Comment by DmitriyLewen** [comment]
Hello @muhammad-asn 
Thanks for your report!

Perhaps `pypi` doesn't have all version.
[repository](https://github.com/huggingface/text-generation-inference) has more tags (including 2.0.0).

 > the library not installed.
 
 Can you scan your image with `-f json --list-all-pkgs` and check `filepath` to  `text-generation` package?

Regards, Dmitriy
